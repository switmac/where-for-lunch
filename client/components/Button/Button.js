import React from 'react';
import PropTypes from 'prop-types';
import classNames from 'classnames/bind';
import styles from './Button.css';

const Button = ({ onClick, children, theme, isDisabled }) => {
  const clickHandler = isDisabled ? null : onClick;
  return (
    <button
      className={classNames({
        [styles.root]: true,
        [styles[theme]]: true,
        disabled: isDisabled,
      })}
      onClick={clickHandler}
    >
      { children }
    </button>
  );
};

Button.propTypes = {
  children: PropTypes.oneOfType([
    PropTypes.string,
  ]),
  onClick: PropTypes.func,
  theme: PropTypes.string,
  isDisabled: PropTypes.bool,
};

export default Button;
