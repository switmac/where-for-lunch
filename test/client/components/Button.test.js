import React from 'react';
import { shallow, mount } from 'enzyme';
import Button from 'client/components/Button/Button';
import renderer from 'react-test-renderer';

test('With Enzyme, Button component is enabled when isDisabled property passed false', () => {
  const isDisabled = false;
  const wrapper = shallow(
    <Button isDisabled={isDisabled} />,
  );
  const p = wrapper.find('.disabled');
  expect(p.length).toBe(0);
});

test('With Enzyme, Button component is enabled when isDisabled property is not passed', () => {
  const wrapper = shallow(
    <Button />,
  );
  const p = wrapper.find('.disabled');
  expect(p.length).toBe(0);
});

test('With Enzyme, Button component is disabled when isDisabled property passed true', () => {
  const isDisabled = true;
  const wrapper = shallow(
    <Button isDisabled={isDisabled} />,
  );
  const p = wrapper.find('.disabled');
  expect(p.length).toBe(1);
});

test('With Jest snapshot, render a disabled button', () => {
  const isDisabled = true;
  const buttonComponent = renderer
    .create(<Button isDisabled={ isDisabled }>Test</Button>)
    .toJSON();
  expect(buttonComponent).toMatchSnapshot();
});

test('With Enzyme, simulate click to an enabled button', () => {
  const isDisabled = false;
  let clicked = false;
  const testFxn = () => {
    clicked = true;
  };
  testFxn.bind(this);
  const buttonComponent = mount(
    <Button isDisabled={ isDisabled } onClick={testFxn}>Test</Button>,
  );
  buttonComponent.simulate('click');
  expect(clicked).toEqual(true);
});


test('With Enzyme, simulate click to a disabled button', () => {
  const isDisabled = true;
  let clicked = false;
  const testFxn = () => {
    clicked = true;
  };
  testFxn.bind(this);
  const buttonComponent = mount(
    <Button isDisabled={ isDisabled } onClick={testFxn}>Test</Button>,
  );
  buttonComponent.simulate('click');
  expect(clicked).toEqual(false);
});
