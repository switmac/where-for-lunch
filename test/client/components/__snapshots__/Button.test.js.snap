// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`With Jest snapshot, render a disabled button 1`] = `
<button
  className="root undefined disabled"
  onClick={null}
>
  Test
</button>
`;
